src/services/formatting/formatting.ts

```diff
@@ -515,17 +515,22 @@ function formatSpanWorker(
         processNode(enclosingNode, enclosingNode, startLine, undecoratedStartLine, initialIndentation, delta);
     }
 
-    if (!formattingScanner.isOnToken()) {

```

Comment:
We really only care if there's any trivia remaining, regardless of whether we ended up on a token. This condition was causing another bug, illustrated in formatSelectionWithTrivia8.ts